clang/lib/Format/UnwrappedLineParser.cpp

```diff
@@ -1134,6 +1134,14 @@ void UnwrappedLineParser::parsePPDefine() {
     return;
   }
 
+  if (Style.IgnorePPDefinitions) {
+    do {
+      nextToken();
+    } while (!eof());
```

Comment:
I assume you have tested this on real code. Can you explain why it is okay to check for `eof()`? I really don't know this code, and am a bit surprised.